src/Mod/Measure/Gui/QuickMeasure.cpp

```diff
@@ -111,24 +111,30 @@ void QuickMeasure::tryMeasureSelection()
 
 bool QuickMeasure::shouldMeasure(const Gui::SelectionChanges& msg) const
 {
-    if (msg.Type == Gui::SelectionChanges::SetPreselect
-        || msg.Type == Gui::SelectionChanges::RmvPreselect) {
-        return false;
-    }
 
+    // measure only IF
     Gui::Document* doc = Gui::Application::Instance->activeDocument();
-    return doc != nullptr;
+    if (doc) {
+        // we have a document
+        if (msg.Type == Gui::SelectionChanges::AddSelection || msg.Type == Gui::SelectionChanges::RmvSelection || msg.Type == Gui::SelectionChanges::SetSelection ||  msg.Type == Gui::SelectionChanges::ClrSelection) {
+            // the event is about a change in selected objects
+            if (Gui::Control().activeDialog()==nullptr) {
+                // we are not in a tool dialog where the user needs to click on stuff
+                return true;
+            }
+        }
+    }
+    return false;
 }
 
 bool QuickMeasure::isObjAcceptable(App::DocumentObject* obj)
 {
-    if (!obj || !obj->isDerivedFrom(Part::Feature::getClassTypeId())) {
-        return false;
+    // only measure shapes
+    if (obj && obj->isDerivedFrom(Part::Feature::getClassTypeId())) {
+        return true;
     }
 
-    std::string vpType = obj->getViewProviderName();
-    auto* vp = Gui::Application::Instance->getViewProvider(obj);
-    return !(vpType == "SketcherGui::ViewProviderSketch" && vp && vp->isEditing());
```

Comment:
the check for "not editing a sketch" is no longer needed, since editing *anything* will now disable QuickMeasure

Reply:
Minor: In this case you can directly `return obj && obj->isDerivedFrom(Part::Feature::getClassTypeId())`

Reply:
IMHO the code is functionally easier to read as is - the compiler will further optimize that when generating machine code anyway

Reply:
I'm fine with leaving it like this, as long as the linter doesn't complain about it -- pylint is much crankier about such constructions than our current C++ linters.